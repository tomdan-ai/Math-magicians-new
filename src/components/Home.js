const Home = () => (
  <div>
    <h2>Welcome to our page!</h2>
    <p>
      Lorem ipsum dolor sit amet consectetur, adipisicing elit. Ullam quam laboriosam
      mollitia porro, est illo aspernatur vel tenetur saepe enim dolores ducimus fugiat
      eveniet dicta explicabo! Dolores perferendis tempora quaerat!
    </p>
    <p>
      Lorem ipsum dolor sit amet consectetur, adipisicing elit. Ullam quam laboriosam
      mollitia porro, est illo aspernatur vel tenetur saepe enim dolores ducimus fugiat
      eveniet dicta explicabo! Dolores perferendis tempora quaerat!
    </p>
  </div>
);

export default Home;
