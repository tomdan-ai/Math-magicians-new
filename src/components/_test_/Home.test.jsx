import { render, screen } from "@testing-library/react";
import Home from "../Home";
import { BrowserRouter } from "react-router-dom";
import renderer from 'react-test-renderer';

describe("Home", () => {
  test("Home is being rendered correctly", () => {
    const container = document.createElement("div");
    render(
      <BrowserRouter basename="/">
        <Home />
      </BrowserRouter>,
      container
    );
    const headings = screen.getAllByRole("heading");
    const paragraphs = screen.getAllByText(/Lorem ipsum/);
    
    expect(headings.length).toBeGreaterThan(0);
    expect(paragraphs.length).toBeGreaterThan(0);
  });
});

describe("Home Snapshot", () => {
  test("Home matches DOM snapshot", () => {
    const tree = renderer
      .create(
        <BrowserRouter basename="/">
          <Home />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
