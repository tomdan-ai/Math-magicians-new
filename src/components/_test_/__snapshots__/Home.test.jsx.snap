// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Snapshot Home matches DOM snapshot 1`] = `
<div>
  <h2>
    Welcome to our page!
  </h2>
  <p>
    Lorem ipsum dolor sit amet consectetur, adipisicing elit. Ullam quam laboriosam mollitia porro, est illo aspernatur vel tenetur saepe enim dolores ducimus fugiat eveniet dicta explicabo! Dolores perferendis tempora quaerat!
  </p>
  <p>
    Lorem ipsum dolor sit amet consectetur, adipisicing elit. Ullam quam laboriosam mollitia porro, est illo aspernatur vel tenetur saepe enim dolores ducimus fugiat eveniet dicta explicabo! Dolores perferendis tempora quaerat!
  </p>
</div>
`;
